google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/DependencyVersionWithBomInspection.java

```diff
@@ -0,0 +1,172 @@
+
+    Set<CloudLibrary> cloudLibraries =
+        CloudLibraryProjectState.getInstance(module.getProject()).getCloudLibraries(module);
+
+    if (cloudLibraries.isEmpty()) {
+      return;
+    }
+
+    if (!CloudLibraryProjectState.getInstance(module.getProject())
+        .getCloudLibraryBom(module)
+        .isPresent()) {
+      return;
+    }
+
+    projectModel
+        .getDependencies()
+        .getDependencies()
+        .forEach(
+            dependency -> {
+              if (hasVersion(dependency) && isCloudLibraryDependency(dependency, cloudLibraries)) {
+                holder.createProblem(
+                    dependency.getVersion(),
+                    HighlightSeverity.GENERIC_SERVER_ERROR_OR_WARNING,
+                    GctBundle.message(
+                        "cloud.libraries.version.with.bom.inspection.problem.description"),
+                    new StripDependencyVersionQuickFix());
+              }
+            });
+  }
+
+  /** Checks to see if the {@link MavenDomDependency} has a version defined. */
+  private boolean hasVersion(MavenDomDependency dependency) {
+    return dependency.getVersion().exists();
+  }
+
+  /**
+   * Checks the supplied {@link MavenDomDependency} to see if it is contained in the known set of
+   * managed Google {@link CloudLibrary cloudLibraries}.
+   *
+   * @param dependency the maven dependency we are checking
+   * @param cloudLibraries the set of {@link CloudLibrary cloudLibraries} configured in the project
+   * @return {@code true} if the maven dependency is a google cloud library, and {@code false}
+   *     otherwise
+   */
+  private boolean isCloudLibraryDependency(
+      MavenDomDependency dependency, Set<CloudLibrary> cloudLibraries) {
+    return cloudLibraries
+        .stream()
+        .anyMatch(
+            library ->
+                CloudLibraryUtils.getFirstJavaClientMavenCoordinates(library)
+                    .map(
+                        coords ->
+                            domValueEquals(dependency.getGroupId(), coords.getGroupId())
+                                && domValueEquals(
+                                    dependency.getArtifactId(), coords.getArtifactId()))
+                    .orElse(false));
+  }
+
+  private boolean domValueEquals(GenericDomValue domValue, @Nullable String value) {
+    return domValue.getStringValue() != null && domValue.getStringValue().equals(value);
+  }
+
+  /**
+   * A {@link LocalQuickFix} that will delete the {@link XmlTag} contained within the {@link
+   * ProblemDescriptor}.
+   */
+  private static class StripDependencyVersionQuickFix implements LocalQuickFix {
+    @Nls
+    @NotNull
+    @Override
+    public String getFamilyName() {
+      return GctBundle.message("cloud.libraries.version.with.bom.quickfix.title");
+    }
+
+    @Override
+    public void applyFix(@NotNull Project project, @NotNull ProblemDescriptor descriptor) {
+      XmlTextImpl xmlElement = (XmlTextImpl) descriptor.getPsiElement();
```

Comment:
`descriptor.getPsiElement()` cannot be null itself in case of some internal structure changes?

Reply:
what do you mean by internal structure changes?

Reply:
It looks like it should not be null since it indicates the issue with the element and should not be called if element itself is gone, but since we use a lot of IJ APIs and they sometime return null unexpectedly, I wonder if this can happen here as well - cannot think of any reason except some internal (IDE) level mismatches in inspection/fix state

Reply:
Yeah. I agree with your reasoning. I'll add a check though just to be safe, and will raise a `log.error` if it happens so we get notified. wdyt?

Reply:
awesome. I start feeling everything could be null coming from an IDE :)